state.go

```diff
@@ -32,6 +36,7 @@ type serializedState struct {
 	SRTPProtectionProfile uint16
 	RemoteCertificate     []byte
 	IsClient              bool
+	ExtendedMasterSecret  bool
```

Comment:
Do we need to serialize this? `ExtendedMasterSecret` doesn't have any effect post-handshake right?

I think the serialize/de-serialize is only used post-handshake. Sorry for being so nitty, just don't want people to depend on any state in future features by accident that we don't actually need to provide.

I already am planning on ripping more out of `Conn` when I get a chance :)

Reply:
Ahh yes makes sense, I will move this out

Reply:
Fixed!